### ISSUE TITLE
Possibly incorrect code sample in manual for Java Config when configuring a job

### ISSUE THREAD

**Issue by alwyn** [spring-projects/spring-batch#3722]
Section 4.2 of the manual for version 4.2.2 of Spring Batch provides an example of creating a custom **BatchConfigurer** by extending **DefaultBatchConfigurer**.

The example extends DefaultBatchConfigurer by creating an anonymous instance and overriding the methods you want to modify.

The issue is that it does so by calling a default constructor on DefaultBatchConfigurer that is protected.  

The only public constructor the class has expects a Datasource and you may not want to provide a datasource if you want to define a memory based repository.

In summary, the example code cannot compile, at least for me.

PS. I am sorry I missed the link where you can change the type of the issue.

**Comment by benas** [comment]
Duplicated by #3888 .
